From c56d9b807fdbbc3f093f34ad80734eb982b2c5aa Mon Sep 17 00:00:00 2001
From: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Date: Fri, 16 Nov 2018 12:17:13 +0900
Subject: [PATCH 373/909] Revert "rcar-vin: rcar-csi2: Add blank margin when
 caluculating bit rate"

commit e90eebf231b4d18b1c0071b317ea2e6e50d0bceb from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This reverts commit 01d619a392ac781da05c4622abcd73b8a76df982.

The reason is follows.
When pixel clock rate with Hblank/Vblank margin added from decoder
side was propageted, this calculation is unnecessary and should be
removed. Also, it is not reasonable to calculate the margin with
the CSI2 driver. The propagating the pixel clock rate that margined
Hblank/Vblank on the decoder driver side is correct.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/rcar-vin/rcar-csi2.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/media/platform/rcar-vin/rcar-csi2.c b/drivers/media/platform/rcar-vin/rcar-csi2.c
index 173fbe33e554..c6b2343bdd6b 100644
--- a/drivers/media/platform/rcar-vin/rcar-csi2.c
+++ b/drivers/media/platform/rcar-vin/rcar-csi2.c
@@ -478,13 +478,6 @@ static int rcsi2_calc_mbps(struct rcar_csi2 *priv, unsigned int bpp)
 	 * bps = link_freq * 2
 	 */
 	mbps = v4l2_ctrl_g_ctrl_int64(ctrl) * bpp;
-
-	/* Vblank's margin is 1.05 times of the horizontal size */
-	mbps = div_u64(mbps * 105, 100);
-
-	/* Hblank's margin is 1.13 times of the vertical size */
-	mbps = div_u64(mbps * 113, 100);
-
 	do_div(mbps, priv->lanes * 1000000);
 
 	return mbps;
-- 
2.17.1

